app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx

```diff
@@ -105,6 +105,7 @@ const FineTuneModal = ({ disclosure }: { disclosure: UseDisclosureReturn }) => {
   const [trainingConfigOverrides, setTrainingConfigOverrides] = useState<
     Partial<AxolotlConfig> | undefined
   >();
+  const [calculationRefetchInterval, setCalculationRefetchInterval] = useState(0);
```

Comment:
Think this could be replaced more elegantly with a `useMemo`.

Reply:
Can it? The data we rely on to determine whether we should be polling or not is fetched from useDatasetTrainingCost, which is also the place where we need to specify our polling interval. I think a combination of useEffect and setState is the best way to get around this circular dependency, but I could be wrong.

Reply:
I'm not sure I understand how to do it.
If I use `useMemo` to handle the state, it will depend on the query response: `price.data?.calculating`. The query response, at the same time, will depend on the interval from `useMemo` state, causing the "variable used before its declaration" error.
Maybe I misunderstood the comment?